src/bp/core/database/tables/bot-specific/dialog_sessions.ts

```diff
@@ -7,7 +7,7 @@ export class DialogSessionTable extends Table {
     let created = false
 
     await this.knex.createTableIfNotExists(this.name, table => {
-      table.string('id').notNullable()
+      table.text('id').notNullable()
```

Comment:
I think it would be better if `id` was explicitly declared as a primary key (using `.primary()`).

On the subject of primary keys, I wasn't sure a column of variable sized `text` was indexable so I did some tests. It seems that postgres is in fact able to create an index for a `text` pk. With a db containing about 2 000 000 sessions there wasn't much of a difference of performance that I could notice between `varchar(255)` and `text` when getting by id (index scan was used for both)

So I think this works for now, but I think it would better in the future if we avoided putting anything in the id that isn't the id itself (sot not `${botId}::${channel}::${target}${threadId ? `::${threadId}` : ''}`). We could just have additional columns for that info, and cache the session instead of extracting the info from the id.

Reply:
> I think it would be better if id was explicitly declared as a primary key (using .primary()).

It is, see line 18 of this file. Or am I missing something here?

> So I think this works for now, but I think it would better in the future if we avoided putting anything in the id that isn't the id itself

I totally agree with you that ids should be kept as simple as possible. The best scenarios being auto-increment or something like GUID/UUID imho!

Reply:
> It is, see line 18 of this file. Or am I missing something here?

You're right I missed that! After looking a bit it seems we're not very consistent in how we declare primary keys. What I meant was declaring it like this :
```ts
table.text('id').primary()
```
We use both methods everywhere so no need to change it after all